File: VitalSphere/VitalSphere.Services/Database/WellnessService.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace VitalSphere.Services.Database
{
    public class WellnessService
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(1000)]
        public string? Description { get; set; }

        public byte[]? Image { get; set; }

        [Required]
        [Range(0.01, 10000)]
        public decimal Price { get; set; }

        /// <summary>
        /// Duration of the service in minutes.
        /// </summary>
        [Range(1, 1440)]
        public int? DurationMinutes { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Required]
        public int WellnessServiceCategoryId { get; set; }
        public WellnessServiceCategory WellnessServiceCategory { get; set; } = null!;

        public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    }
}

File: VitalSphere/UI/vital_sphere_mobile/lib/model/wellness_service.dart
import 'package:json_annotation/json_annotation.dart';

part 'wellness_service.g.dart';

@JsonSerializable()
class WellnessService {
  final int id;
  final String name;
  final String? description;
  final String? image;
  final double price;
  final int? durationMinutes;
  final bool isActive;
  final DateTime createdAt;
  final int wellnessServiceCategoryId;
  final String wellnessServiceCategoryName;

  WellnessService({
    this.id = 0,
    this.name = '',
    this.description,
    this.image,
    this.price = 0.0,
    this.durationMinutes,
    this.isActive = true,
    required this.createdAt,
    this.wellnessServiceCategoryId = 0,
    this.wellnessServiceCategoryName = '',
  });

  factory WellnessService.fromJson(Map<String, dynamic> json) =>
      _$WellnessServiceFromJson(json);
  Map<String, dynamic> toJson() => _$WellnessServiceToJson(this);
}


